utils/collect-uncommitted/lib/collect-uncommitted.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+const chalk = require("chalk");
+const { exec, execSync } = require("@lerna/child-process");
+
+module.exports = collectUncommitted;
+module.exports.sync = sync;
+
+const maybeColorize = colorize => s => (s !== " " ? colorize(s) : s);
+const cRed = maybeColorize(chalk.red);
+const cGreen = maybeColorize(chalk.green);
+
+const replaceStatus = (_, maybeGreen, maybeRed) => `${cGreen(maybeGreen)}${cRed(maybeRed)}`;
+
+const colorizeStats = stats =>
+  stats.replace(/^([^U]| )([A-Z]| )/gm, replaceStatus).replace(/^\?{2}|U{2}/gm, cRed("$&"));
+
+const splitOnNewLine = (string = "") => string.split("\n");
+
+const filterEmpty = (strings = []) => strings.filter(s => s.length !== 0);
+
+const o = (l, r) => x => l(r(x));
+
+const transformOutput = o(filterEmpty, o(splitOnNewLine, colorizeStats));
+
+function collectUncommitted(options = {}) {
+  return exec("git", "status -s", options).then(transformOutput);
```

Comment:
This module looks fine as-is, though I am a little concerned that the `-s` option of `git status` doesn't guarantee stability across versions (`--porcelain` [v1](https://www.git-scm.com/docs/git-status#_porcelain_format_version_1) claims to do that, otherwise identical to the short format with colors always off and paths always relative).

Anyway, we can cross that bridge when we come to it, if at all.